### PR TITLE
Create model classes

### LINKED ISSUE #7: Developing JPA Entity classes
**Is your feature request related to a problem? Please describe.**
JPA entity classes have to be added to handle the data. Which will then be use for backend logic.

**Describe the solution you'd like**
Create Classes with the specific annotations using the designed class diagram: https://drive.google.com/file/d/13BsmmxXJBgv3Ul5khC2Isac3Z7IxdfOU/view?usp=sharing

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Gimhan-minion**
## Purpose
The purpose of this PR is to fix #7 

## Goals
To create the classes according to the class diagram.

## Approach
Created the model package with the necessary classes.

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related Links
https://drive.google.com/file/d/13BsmmxXJBgv3Ul5khC2Isac3Z7IxdfOU/view

### CODE REVIEW

File: api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.sefglobal.scholarx.model;
+
+import java.util.Date;
+
+public class BaseScholarxModel {
+    private String id;
```
Comment (YohanAvishke) [inline api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java:6]:
```suggestion
    private int id;
```

File: api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package org.sefglobal.scholarx.model;
+
+import java.util.Date;
+
+public class BaseScholarxModel {
```
Comment (Piumal1999) [inline api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java:5]: This should be `public abstract class BaseScholarxModel implements Serializable` right? @YohanAvishke

Comment (Piumal1999) [inline api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java:5]: This should be `public abstract class BaseScholarxModel implements Serializable` right? @YohanAvishke

File: api/src/main/java/org/sefglobal/scholarx/model/Profile.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
+package org.sefglobal.scholarx.model;
+
+import org.sefglobal.scholarx.util.ProfileType;
+
+public class Profile {
```
Comment (YohanAvishke) [inline api/src/main/java/org/sefglobal/scholarx/model/Profile.java:5]: extend from `BaseScholarxModel `

File: api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java (hunk 2 above)
Comment (YohanAvishke) [reply api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java:5]: yes, nice find @Piumal1999

Comment (Gimhan-minion) [reply api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java]: Done

File: api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java (hunk 1 above)
Comment (Gimhan-minion) [reply api/src/main/java/org/sefglobal/scholarx/model/BaseScholarxModel.java]: Done

File: api/src/main/java/org/sefglobal/scholarx/model/Profile.java (hunk 3 above)
Comment (Gimhan-minion) [reply api/src/main/java/org/sefglobal/scholarx/model/Profile.java]: Done

File: api/src/main/java/org/sefglobal/scholarx/model/EnrolledUser.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package org.sefglobal.scholarx.model;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.sefglobal.scholarx.util.EnrolmentState;
+
+@Getter @Setter
+public class EnrolledUser extends BaseScholarxModel{
```
Comment (jayasanka-sack) [inline api/src/main/java/org/sefglobal/scholarx/model/EnrolledUser.java:8]: Shall we make this class abstract?

**Comment by jayasanka-sack** [comment]
Shall we avoid using Lombok getters and setters?

File: api/src/main/java/org/sefglobal/scholarx/model/EnrolledUser.java (hunk 4 above)
Comment (Gimhan-minion) [reply api/src/main/java/org/sefglobal/scholarx/model/EnrolledUser.java]: done

**Pull request merged**